### ISSUE TITLE
Takes context as a parameter but doesn't do anything with it

### ISSUE THREAD

**Issue by ciokan** [sethvargo/go-limiter#35]
Looks like a false trap that context that `Take` takes as a parameter.

**Comment by ciokan** [comment]
Nevermind, I see that `Take` does not wait at all so the ctx is useless in this case.

**Comment by ciokan** [comment]
True. I thought `Take` waits until it can "give" as in other packages and I was looking for a way to stop it with a context in case it waits too much but your implementation does not wait so it's all good.

**Comment by sethvargo** [comment]
Ah - yes. `Take` immediately returns. The boolean returned indicates whether the take was successful. This is documented in the godoc: https://github.com/sethvargo/go-limiter/blob/main/store.go#L34-L35
